documentation/player-portal/web/v2/api.md

```diff
@@ -0,0 +1,1886 @@
+
+##### PARAMETERS:
+
+* **targetId** String _( Optional )_ The DOM player target id attribute string. ( if not included, you must include targetId in "settings" object
+* **settings** [kWidget.settingsObject][1] Object of settings to be used in embedding.
+
+##### EXAMPLES:
+
+* * *
+
+#### kWidget.getKalturaThumbUrl (settings)
+
+Get video thumbnail URL.
+
+* * *
+
+#### kWidget.addReadyCallback (readyCallback)
+
+Adds a ready callback to be called after the KDP or HTML5 player is ready.
+
+##### PARAMETERS:
+
+* **readyCallback** String Function to call after a player or widget is ready on the page.
+
+##### EXAMPLES:
+
+* * *
+
+#### kWidget.destroy (target)
+
+Removes the player from the DOM.
+
+##### PARAMETERS:
+
+* **target** String The target element or element ID to destroy.
+
+##### EXAMPLES:
+
+###  User Agent and Feature Detection 
+
+* * *
+
+#### kWidget.isMobileDevice ()
+
+If user agent identifies as mobile device ( android, iOS or windows phone )
+
+* * *
+
+#### kWidget.supportsHTML5 ()
+
+If the device or browser supports HTML5
+
+* * *
+
+#### kWidget.supportsFlash ()
+
+If the device or browser supports Flash ( version 10 and above )
+
+* * *
+
+#### kWidget.isIOS ()
+
+If user agent identifies as iOS device
+
+* * *
+
+#### kWidget.isIE ()
+
+If user agent identifies as Internet Explorer browser
+
+* * *
+
+#### kWidget.isIE8 ()
+
+If user agent identifies as Internet Explorer 8 browser
+
+* * *
+
+#### kWidget.isAndroid ()
+
+If user agent identifies as Android device
+
+* * *
+
+#### kWidget.isWindowsDevice ()
+
+If user agent identifies as Windows Phone device
+
+###  Settings Embed Object: 
+
+* * *
+
+#### kWidget.settingsObject
+
+* **targetId** String The DOM player target id attribute string if not defined as top level param.
+* **wid** String Widget id, usually the partner id prefixed by underscore.
+* **uiconf_id** Number _( Optional )_ The player uiconf_id
+* **entry_id** String The content entry id. Can be left empty for a JavaScript based entry id.
+* **flashvars** Object Runtime configuration object, can override arbitrary uiVars and plugin config.
+* **params** Object Runtime configuration object, can override arbitrary uiVars and plugin config.
+* **cache_st** String _( Optional )_ String to burst player cache
+* **readyCallback** Function _( Optional )_ Local callback method to be called once player is ready for bindings. Player id is passes as an argument. See [live example][2]
+
+### Server API requests ( kWidget.api )
+
+kWidget Server API enables direct [Kaltura Server API][3] calls from JavaScript. This should not be confused with the [JavaScript client library][4], which offers object mappings and works with the code generated in the [test me console][5].   
+The Kaltura Server API offers minimal object validation, in exchange for being much smaller, and included with every kaltura player library include.
+
+Creating a kWidget API object, issue a playlist request, log the result: 
```

Comment:
All code samples should be properly marked as such to ensure proper highlighting.